### PR TITLE
Ignore Failed Submit Jobs When Kicking Off Next Jobs (Form526)

### PR DESCRIPTION
**Pull request by edmangimelli**
With the new [_multiple BIRLS IDs flow_](https://github.com/department-of-veterans-affairs/vets-api/pull/5245), a Form526Submission might have 1 or more failed SubmitForm526AllClaim jobs before having a successful one. Consequently, _all jobs succeeded_ must be measured differently now (to ensure ancillary jobs get kicked off).

_Instead of:_
`.all?(&:success?)`
_now:_
at least one successful `SubmitForm526AllClaim` ➕ if there are any other types of jobs (non-submit jobs), they all succeeded

### CODE REVIEW

File: app/models/form526_submission.rb [hunk 1]
```diff
@@ -205,7 +206,29 @@ def birls_id=(value)
...
+    successful = submit_form_526_job_statuses.where(status: 'success').load
+    warn = ->(message) { log_message_to_sentry(message, :warn, { form_526_submission_id: id }) }
+    warn.call 'There are multiple successful SubmitForm526 job statuses' if successful.size > 1
+    if successful.size == 1 && submit_form_526_job_statuses.last.unsuccessful?
+      warn.call "There is a successful SubmitForm526 job, but it's not the most recent SubmitForm526 job"
+    end
+  end
+
+  def all_other_jobs_succeeded_if_any?
+    form526_job_statuses.where.not(job_class: SUBMIT_FORM_526_JOB_CLASSES).all?(&:success?)
```
Comment (edmangimelli) [inline app/models/form526_submission.rb:231]: can be vacuously true

File: app/models/form526_submission.rb [hunk 2]
```diff
@@ -205,7 +206,29 @@ def birls_id=(value)
...
+
+  def jobs_succeeded?
+    a_submit_form_526_job_succeeded? && all_other_jobs_succeeded_if_any?
+  end
+
+  class SubmitForm526JobStatusesError < StandardError; end
+
+  def a_submit_form_526_job_succeeded?
+    submit_form_526_job_statuses = form526_job_statuses.where(job_class: SUBMIT_FORM_526_JOB_CLASSES).order(:updated_at)
+    submit_form_526_job_statuses.presence&.any?(&:success?)
```
Comment (edmangimelli) [inline app/models/form526_submission.rb:220]: cannot be vacuously true

**Review by thilton-oddball — APPROVED** [review]
Looks good

**Review by annaswims — APPROVED** [review]
LGTM

**Pull request merged**